syntax = "proto3";

message DtuInfoMO {
    int32 device_kind = 1;
    int32 dtu_sw = 2;
    int32 dtu_hw = 3;
    int32 dtu_step_time = 4;
    int32 dtu_rf_hw = 5;
    int32 dtu_rf_sw = 6;
    int32 access_model = 7;
    string gprs_vsn = 8;
    string wifi_vsn = 9;
    string ka_nub = 10;
    int32 dtu_rule_id = 11;
    int32 dtu_error_code = 12;
    int32 grid_type = 13;
    int32 zero_export_switch= 14;
    int32 surplus_power_a = 15;
    int32 surplus_power_b = 16;
    int32 surplus_power_c = 17;
    int32 zero_export_control= 18;
    int32 phase_balance_switch = 19;
    int32 tolerance_between_phases = 20;
}

message MeterInfoMO {
    int32 device_kind = 1;
    int64 meter_sn = 2;
    int32 meter_model = 3;
    int32 meter_ct = 4;
    int32 com_way = 5;
    int32 access_mode = 6;
}

message RpInfoMO {
    int32 device_kind = 1;
    int64 rp_sn = 2;
    int32 rp_sw = 3;
    int32 rp_hw = 4;
    int32 rp_rule_id = 5;
}

message PvInfoMO {
    int32 device_kind = 1;
    int64 pv_sn = 2;
    int32 pv_usfw = 3;
    int32 pv_sw = 4;
    int32 pv_hw_pn = 5;
    int32 pv_hw = 6;
    int32 pv_gpf_code = 7;
    int32 pv_gpf = 8;
    int32 pv_rf_hw = 9;
    int32 pv_rf_sw = 10;
    int32 mi_rule_id = 11;
}

message FeatureMO {
    int32 key = 1;
    string value = 2;
}

message InfoDataReqDTO {
    string dtu_sn = 1;
    int32 time = 2;
    int32 device_nub = 3;
    int32 pv_nub = 4;
    int32 package_nub = 5;
    int32 package_now = 6;
    int32 channel = 7;
    DtuInfoMO mDtuInfo = 8;
    repeated MeterInfoMO mMeterInfo = 9;
    repeated RpInfoMO mRpInfo = 10;
    repeated PvInfoMO mpvInfo = 11;
    repeated FeatureMO m_feature = 12;
}

message InfoDataResDTO {
    string time_ymd_hms = 1;
    int32 offset = 2;
    int32 package_now = 3;
    int32 error_code = 4;
    int32 time = 5;
}
